src/qml/filters/gradientmap/ui_gradientmap.qml

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2017-2022 Meltytech, LLC
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import QtQuick
+import QtQuick.Controls
+import QtQuick.Layouts
+import Shotcut.Controls as Shotcut
+
+Item {
+    property var defaultParameters: ['stop.1', 'stop.2']
+    property bool _disableUpdate: true
+
+    function setControls() {
+        _disableUpdate = true;
+        fgGradient.parseStops(filter.getGradient('stop'));
+        _disableUpdate = false;
+    }
+
+    width: 350
+    height: 180
+    Component.onCompleted: {
+        if (filter.isNew) {
+            filter.set('stop.1', '#ff000000 0.0');
+            filter.set('stop.2', '#ffffffff 1.0');
+            filter.savePreset(defaultParameters);
+        }
+        setControls();
+    }
+
+    ColumnLayout {
+        // columns: 5
```

Comment:
The layout seems weird. The "Gradient Stops" label usually appears left of its controls, but I think here you are trying to say everything below here are the gradient stops like a heading. Actually, for this UI since there are so few parameters maybe that label is not needed. It should probably be switched to a `GridLayout` with the addition of Presets.